plugins/org.jboss.tools.jst.web.ui/src/org/jboss/tools/jst/web/ui/internal/editor/outline/CategoryProviderFactory.java

```diff
@@ -39,26 +45,34 @@ public static CategoryProviderFactory getInstance() {
 	 * As soon as other providers appear, classes will be obtained from an extension point. 
 	 */
 	private CategoryProviderFactory() {
-		Bundle b = Platform.getBundle("org.jboss.tools.jst.web.ui");
-		if(b != null) {
-			try {
-				Class<?> c = b.loadClass("org.jboss.tools.jst.web.ui.internal.properties.angular.AngularJSPropertySetViewer");
-				knownDescriptors.add(new CategoryDescriptor("AngularJS", "angularJS", AngularCategoryFilter.class, c));
-			} catch (ClassNotFoundException e) {
-			}
-			try {
-				Class<?> c = b.loadClass("org.jboss.tools.jst.web.ui.internal.properties.jquery.JQueryPropertySetViewer");
-				knownDescriptors.add(new CategoryDescriptor("jQuery", "jQueryMobile",  JQueryCategoryFilter.class, c));
-			} catch (ClassNotFoundException e) {
-			}
-			try {
-				Class<?> c = b.loadClass("org.jboss.tools.jst.web.ui.internal.properties.html.HTMLPropertySetViewer");
-				knownDescriptors.add(new CategoryDescriptor("HTML", "htmlFile", c));
-			} catch (ClassNotFoundException e) {
-			}
+		IExtensionPoint point = Platform.getExtensionRegistry().getExtensionPoint(POINT_ID);
+		for (IConfigurationElement element : point.getConfigurationElements()) {
+			String name = element.getAttribute("name"); //$NON-NLS-1$
+			String uri = element.getAttribute("uri"); //$NON-NLS-1$
+			Class<? extends ICategoryFilter> categoryFilterClass = getCategoryFilterClass(element);
+			Class<? extends IPropertySetViewer> propertySetViewerClass = getPropertySetViewerClass(element);
+			knownDescriptors.add(new CategoryDescriptor(name, uri,  categoryFilterClass, propertySetViewerClass));
```

Comment:
Is name or url is required? If yes and any of them should not be null/empty then we should check if the value of the corresponding attribute is set/correct and log if not. Otherwise in case of broken extension it can cause properties view crashing.